recipes/glm/all/conandata.yml

```diff
@@ -5,8 +5,12 @@ sources:
   "0.9.9.6":
     url: "https://github.com/g-truc/glm/archive/0.9.9.6.tar.gz"
     sha256: "30b05f31f5d8528caa9fc8fe8132e5a0444d55b1c39db53fe4db8857654afafc"
+  "0.9.9.7":
+    url: "https://github.com/g-truc/glm/archive/0.9.9.7.tar.gz"
+    sha256: "2ec9e33a80b548892af64fbd84a947f93f0e725423b1b7bec600f808057a8239"
 patches:
   "0.9.9.5":
     - patch_file: "patches/0001-Remove-architecture-check-from-CMake-package.patch"
       base_path: "source_subfolder"
   "0.9.9.6": []
+  "0.9.9.7": []
```

Comment:
this is not required, not even the previous line is required. We can safely remove them

Reply:
fixed but it fails on Jenkins...
```
ERROR: glm/0.9.9.6: Error in source() method, line 83
	for patch in self.conan_data["patches"][self.version]:
	KeyError: 'patches'
```

Reply:
oh, now I see what happens. Could you please add a confition before this line https://github.com/conan-io/conan-center-index/blob/fc2de4c42914eb34d90ba417dbe9589fe358e230/recipes/glm/all/conanfile.py#L83 ? We should check if patches is in a key in the dict first to avoid this error

Reply:
I can add a condition, but why does it work locally without condition?
Boost recipe does the same thing without condition, and it also has versions with and without patches in the same condata.yml

Reply:
It fails with the conan-center hook activated, which is something our CI has activated by default to perform some checks and optimizations. I'd recommend you to use as it is very convenient to develop the recipes

Reply:
It is activated on my side, but I think locally it was still using my old conandata.yml during conan create :/
Anyway I was able to reproduce the bug on my side after a conan remove, and it's not related to hooks, just a python error.

Reply:
glad to know. Thanks for testing!